### PR TITLE
TemplateData in match method shouldnt be required as original version's way

### PR DESCRIPTION
**Pull request by jfkz**
I found your module very useful and add one tune from myself: as in original version match has to have 1 required parameter and 1 optional, for compatibility reasons.
I fix this way, pls review.

### CODE REVIEW

File: source/i18n.ts [hunk 1]
```diff
@@ -166,8 +166,8 @@ function compileTemplates(root: Readonly<Record<string, string>>): RepositoryEnt
 
 /* eslint-disable @typescript-eslint/prefer-readonly-parameter-types */
 
-export function match(resourceKey: string, templateData: Readonly<TemplateData>): (text: string, ctx: TelegrafContextWithI18n) => string[] | null {
-  return (text, ctx) => (text && ctx?.i18n && text === ctx.i18n.t(resourceKey, templateData)) ? [text] : null
+export function match(resourceKey: string, templateData?: Readonly<TemplateData>): (text: string, ctx: TelegrafContextWithI18n) => string[] | null {
+  return (text, ctx) => (text && ctx?.i18n && text === ctx.i18n.t(resourceKey, templateData || {})) ? [text] : null
```
Comment (EdJoPaTo) [inline source/i18n.ts]: the `ctx.i18n.t()` method already has an optional second argument so I would prefer using its way of handling the not provided templateData. So the match keeps to be a shortcut and `ctx.i18n.t()` knows how to deal with `undefined` as second argument.

https://github.com/EdJoPaTo/telegraf-i18n/blob/d1e03688f6ad76ae4b4a0cc58308a29c62304c1f/source/context.ts#L40

```suggestion
  return (text, ctx) => (text && ctx?.i18n && text === ctx.i18n.t(resourceKey, templateData)) ? [text] : null
```

**Review by EdJoPaTo — CHANGES_REQUESTED** [review]
Makes sense!
I never used `match` or `reply` so thanks for taking a look!

File: source/i18n.ts (hunk 1 above)
Comment (jfkz) [reply source/i18n.ts]: Yeah, ofc. I did only quick look over the code, so you can modify my commits as you wish.

**Comment by EdJoPaTo** [comment]
Thanks for your time checking out the PR and providing the fix!

**Pull request merged**